Bug: Cloudflare Python Worker throws 500 Error on all /api/* routes (Missing fetch entrypoint)

When running the local development server, querying any backend API endpoint results in a 500 Internal Server Error. The underlying Cloudflare Python runtime crashes because the worker is using the deprecated on_fetch global function instead of the modern class-based structure.

Steps to Reproduce:
- Follow the local setup guide and run, npm run db:init.
- Start the local server with npm run dev.
- Open a browser and navigate to http://localhost:8787/api/stats or http://localhost:8787/api/leaderboard.
- Check the Wrangler terminal logs.

Expected Behavior
The API should return a 200 OK response with the queried database results.

Actual Behavior
The browser receives a 500 Internal Server Error. The Wrangler terminal logs the following fatal crash:
✘ [ERROR] Uncaught Error: Handler does not export a fetch() function.

Proposed Fix
I have diagnosed this locally. Migrating the entry point in workers/main.py from the global async def on_fetch function to the modern class Default(WorkerEntrypoint): syntax completely resolves the crash. I will open a PR with this fix shortly!

--- a/workers/main.py
+++ b/workers/main.py
@@ -2,6 +2,7 @@ from js import Response, Headers, URL
 import json
 import hashlib
 from datetime import datetime
+from workers import WorkerEntrypoint #Added this new import to use the modern class-based structure for Cloudflare Workers. 
 
 # ===================================
 # Configuration
@@ -432,12 +433,13 @@ async def route_request(request, env):
 # Main Entry Point
 # ===================================
 
-async def on_fetch(request, env):
-    """Main entry point for Cloudflare Worker"""
-    try:
-        return await route_request(request, env)
-    except Exception as e:
-        return create_response({
-            'error': 'Internal server error',
-            'message': str(e)
-        }, status=500, origin=request.headers.get('Origin'))
+class Default(WorkerEntrypoint):
+    async def fetch(self, request):
+        """Main entry point for Cloudflare Worker"""
+        try:
+            return await route_request(request, self.env)
+        except Exception as e:
+            return create_response({
+                'error': 'Internal server error',
+                'message': str(e)
+            }, status=500, origin=request.headers.get('Origin'))

